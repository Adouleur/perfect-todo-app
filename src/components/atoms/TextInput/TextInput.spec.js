import { mount } from '@vue/test-utils';
import { createStore } from 'vuex';
import { describe, it, expect, vi } from 'vitest';
import MyComponent from '@/components/atoms/TextInput/TextInput.vue';

describe('MyComponent.vue', () => {
  let store;
  let actions;
  let wrapper;

  beforeEach(() => {
    actions = {
      addTodo: vi.fn()
    };

    store = createStore({
      actions,
      state: {
        todos: []
      }
    });
  });
  wrapper = mount(MyComponent, {
    global: {
      plugins: [store]
    }
  });

  it('should render the input field correctly', () => {
    const input = wrapper.find('input');

    expect(input.exists()).toBe(true);
    expect(input.element.placeholder).toBe('Enter a new task');
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
