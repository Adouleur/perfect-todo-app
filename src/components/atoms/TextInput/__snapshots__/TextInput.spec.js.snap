// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyComponent.vue > matches the snapshot 1`] = `"<input data-v-bcaf8368="" type="text" placeholder="Enter a new task" class="" data-cy="add-task">"`;
